src/nvim/ex_docmd.c

```diff
@@ -8552,7 +8552,12 @@ eval_vars (
         /* Still need to turn the fname into a full path.  It is
          * postponed to avoid a delay when <afile> is not used. */
         autocmd_fname_full = TRUE;
-        result = (char_u *)FullName_save((char *)autocmd_fname, FALSE);
+        if (has_event(EVENT_CMDLINEENTER)
+            || has_event(EVENT_CMDLINELEAVE)) {
+          result = (char_u *)tv_dict_get_string(get_vim_var_dict(VV_EVENT), "cmdtype", true);
+        } else {
+          result = (char_u *)FullName_save((char *)autocmd_fname, FALSE);
+        }
         xfree(autocmd_fname);
```

Comment:
@bfredl does this change make sense to you?

@janlazo  BTW this block was changed on master, needs a rebase.